aws/internal/service/workspaces/waiter/status.go

```diff
@@ -0,0 +1,43 @@
+package waiter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func DirectoryRefreshStateFunc(conn *workspaces.WorkSpaces, directoryID string) resource.StateRefreshFunc {
```

Comment:
FYI I updated the `waiter` package proposal to include more information about naming conventions: https://github.com/terraform-providers/terraform-provider-aws/issues/12840 e.g. `DirectoryState` in this case. 👍  If you have feels on changing those naming conventions please do drop a note!